workspace/__ardulib__/Graphics/XGraphics.h

```diff
@@ -0,0 +1,458 @@
+
+#ifndef X_GRAPHICS_H
+#define X_GRAPHICS_H
+
+#include "XRender.h"
+#include "XVector2.h"
+
+using xod::XColor;
+
+class XFill {
+private:
+public:
+    XFill() {}
+    virtual XColor colorAtPoint(XVector2<int16_t> point) {}
+    virtual void setSolidColor(XColor _color) {}
+};
+
+class XSolidFill : public XFill {
+private:
+    XColor _color;
+public:
+    XSolidFill(XColor color) {
+        _color = color;
+    }
+
+    XColor colorAtPoint(XVector2<int16_t> point);
+    void setSolidColor(XColor _color);
+};
+
+XColor XSolidFill::colorAtPoint(XVector2<int16_t> point) {
+    return _color;
+}
+
+void XSolidFill::setSolidColor(XColor color) {
+    _color = color;
+}
+
+class XBitmap {
+protected:
+    XBitmap* _parent;
+public:
+    XBitmap() {
+        _parent = nullptr;
+    }
+
+    virtual const XFill* getFill() const;
+    virtual const XColor* getBgColor() const;
+    virtual const XColor* getStrokeColor() const;
+
+    bool renderLine(XRender* renderer, int16_t line, uint16_t* buffer);
+    virtual bool renderBitmapLine(XRender* renderer, int16_t line, uint16_t* buffer) { return false; }
```

Comment:
A bad naming. Only reading of the implementation makes clear the difference.
What do you think about renaming them to `renderLineRecursively` and `renderLine`? 🤔